app/api/route/schemas/user_schemas.py

```diff
@@ -52,3 +53,19 @@ class UserSchema(marshmallow.Schema):
     # Output only fields in addition to id field
     created_at = fields.DateTime(dump_only=True)
     updated_at = fields.DateTime(dump_only=True)
+
+
+# Validate input according to the UserSchema
+# and convert input into User
+class UserPostParamsSchema(UserSchema):
+    @marshmallow.post_load
+    def make_user_post_params(self, data: dict, **kwargs: dict[str, Any]) -> User:
+        return User(**data)
```

Comment:
It might make more sense to return a `UserPostParams` object here instead of a `User`. Thoughts? I kept it this way for now so that I could keep changes as minimal as possible, and not have to change much in `create_user`.

Reply:
Not necessary for this endpoint, but how would you implement this if the POST body didn't match the structure of the DB model? I think we'd want an intermediate like the UserPostParams you're suggesting, that way we can also keep the DB model creation/updates down in the persistence layer.